packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -286,9 +356,9 @@ export default function VisualEditor( { styles } ) {
 								className={
 									isTemplateMode
 										? 'wp-site-blocks'
-										: `${ blockListLayoutClass } wp-block-post-content` // Ensure root level blocks receive default/flow blockGap styling rules.
```

Comment:
I was wondering if we might still need the `wp-block-post-content` classname to be added, so that we can hook into styles that are set at the `core/post-content` block in global styles? (It looks like the layout classnames don't output the block classname, so I think we might still need to add it in manually 🤔)

Reply:
I'm a bit hazy on how the global styles stuff works in the post editor, so if you think it's needed I'll just add it back. Won't do any harm to have the class there 😄

Reply:
> I'm a bit hazy on how the global styles stuff works in the post editor

I was looking at this the other day while reviewing other PRs, so in case the breadcrumbs here are helpful: it's pretty naive when it comes to global styles. The server-rendered global styles are added to the block editor settings in PHP [here](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.1/block-editor-settings.php#L77), and then the post editor figures out whether or not to use them [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/editor.js#L167). The styles are then passed down to the post editor's `Layout` component [on this line](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/editor.js#L201). That component then passes the styles to `VisualEditor` [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/components/layout/index.js#L244). And in `VisualEditor`, the styles are passed to `MaybeIframe` which finally passes it along to `EditorStyles`, which ultimately outputs styles in a `style` tag on this line: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/editor-styles/index.js#L81

The TL;DR is that it's _nearly_ as if the global styles stylesheet had been enqueued, only those styles get transformed a bit by the `EditorStyles` component to play nicely in the editor.

I mostly just wanted to write that out somewhere because it finally clicked for me how it's hooked together 😀

Reply:
Thanks, that's really helpful!